### PR TITLE
Vector-of-Counts sketch operator for frequency dedupe.

### PR DESCRIPTION
**Pull request by huangxichen1**

### CODE REVIEW

**Review by jiayu-google — APPROVED** [review]
LGTM!

File: src/estimators/tests/vector_of_counts_sketch_operator_test.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+    self.assert_sketch_equal = _assert_sketches_equal
+
+  @parameterized.parameters(
+      ([1, 0], [0, 1], [1, 1]),
+      (None, [0, 1], [0, 1]),
+      ([1, 0], None, [1, 0]),
+      (None, None, None),
+  )
+  def test_union_works(self, this_stats, that_stats, expected):
```
Comment (kungfucraig) [inline src/estimators/tests/vector_of_counts_sketch_operator_test.py:60]: You could probably drop "_works" from the test names.

File: src/estimators/vector_of_counts_sketch_operator.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Sketch operator for Vector-of-Counts."""
+import copy
+
+
+class SketchOperator:
+  """Sketch operations for supporting frequency dedupe.
+
+  The operator defined in this class is NOT intended to be used as a generic
```
Comment (kungfucraig) [inline src/estimators/vector_of_counts_sketch_operator.py:21]: grammar: The operators defined in this class are NOT intended to be used ....

File: src/estimators/vector_of_counts_sketch_operator.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  sketch operator for the Vector-of-Counts (VoC hereafter). It is ONLY intended
```
Comment (kungfucraig) [inline src/estimators/vector_of_counts_sketch_operator.py:22]: grammar: They are ONLY intended

File: src/estimators/vector_of_counts_sketch_operator.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  to be worked with the stratefied_sketch.py for estimating the frequency. The
```
Comment (kungfucraig) [inline src/estimators/vector_of_counts_sketch_operator.py:23]: grammar: to be used with stratefied_sketch.py for estimation frequency.

File: src/estimators/vector_of_counts_sketch_operator.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  reason behind is that the union operators defined here is for VoCs whose
```
Comment (kungfucraig) [inline src/estimators/vector_of_counts_sketch_operator.py:24]: grammar: reason behind this is that the union operators defined here are for VoCs whose

File: src/estimators/vector_of_counts_sketch_operator.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  underlying sets are DISJOINT, and the difference operator is for those sets
+  that are fully-overlapping.
```
Comment (kungfucraig) [inline src/estimators/vector_of_counts_sketch_operator.py:26]: You might consider adding a qualifier to the class name to help reinforce this.

**Review by kungfucraig — APPROVED** [review]

File: src/estimators/tests/vector_of_counts_sketch_operator_test.py (hunk 1 above)
Comment (huangxichen1) [reply src/estimators/tests/vector_of_counts_sketch_operator_test.py]: Done.

File: src/estimators/vector_of_counts_sketch_operator.py (hunk 2 above)
Comment (huangxichen1) [reply src/estimators/vector_of_counts_sketch_operator.py]: Done.

File: src/estimators/vector_of_counts_sketch_operator.py (hunk 3 above)
Comment (huangxichen1) [reply src/estimators/vector_of_counts_sketch_operator.py]: Done.

File: src/estimators/vector_of_counts_sketch_operator.py (hunk 4 above)
Comment (huangxichen1) [reply src/estimators/vector_of_counts_sketch_operator.py]: Done.

File: src/estimators/vector_of_counts_sketch_operator.py (hunk 5 above)
Comment (huangxichen1) [reply src/estimators/vector_of_counts_sketch_operator.py]: Done.

File: src/estimators/vector_of_counts_sketch_operator.py (hunk 6 above)
Comment (huangxichen1) [reply src/estimators/vector_of_counts_sketch_operator.py]: Done.

**Review by AaronRanAn — APPROVED** [review]
LGTM, thanks Xichen!

File: src/estimators/vector_of_counts_sketch_operator.py [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+    Args:
+      this: a VectorOfCounts.
+      that: a VectorOfCounts, which is assumed to be a subset of this sketch.
+
+    Returns:
+      A VectorOfCounts that is the difference of the input sketches.
+    """
+    if this is None or that is None:
+      return this
```
Comment (my2457) [inline src/estimators/vector_of_counts_sketch_operator.py:90]: don't need deepcopy?

**Review by my2457 — APPROVED** [review]

File: src/estimators/vector_of_counts_sketch_operator.py (hunk 7 above)
Comment (huangxichen1) [reply src/estimators/vector_of_counts_sketch_operator.py]: Thanks for catching this! Now it returns a copy.

**Pull request merged**